cargo install cargo-component error

Trying to install cargo-component:

System details:

```bash
Ubuntu x86_64 GNU/Linux
```

Version details:

```bash
$ rustup --version
rustup 1.27.1 (54dd3d00f 2024-04-24)
$ rustc --version
rustc 1.78.0 (9b00956e5 2024-04-29)
$ cargo --version
cargo 1.78.0 (54d8815d0 2024-03-26)
```
Installation failure:

```bash
$ cargo install cargo-component
Compiling cargo-component-core v0.11.0
error[E0061]: this method takes 1 argument but 0 arguments were supplied
   --> /home/tpmccallum/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-component-core-0.11.0/src/registry.rs:882:38
    |
882 |                 .load_package(client.get_warg_registry(), e.key())
    |                                      ^^^^^^^^^^^^^^^^^-- an argument of type `&str` is missing
    |
note: method defined here
   --> /home/tpmccallum/.cargo/registry/src/index.crates.io-6f17d22bba15001f/warg-client-0.4.2/src/lib.rs:118:18
    |
118 |     pub async fn get_warg_registry(
    |                  ^^^^^^^^^^^^^^^^^
help: provide the argument
    |
882 |                 .load_package(client.get_warg_registry(/* &str */), e.key())
    |                                                       ~~~~~~~~~~~~

   Compiling shell-escape v0.1.5
error[E0308]: mismatched types
   --> /home/tpmccallum/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-component-core-0.11.0/src/registry.rs:882:31
    |
882 |                 .load_package(client.get_warg_registry(), e.key())
    |                  ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Option<&RegistryDomain>`, found future
    |                  |
    |                  arguments to this method are incorrect
    |
note: method defined here
   --> /home/tpmccallum/.cargo/registry/src/index.crates.io-6f17d22bba15001f/warg-client-0.4.2/src/storage.rs:111:14
    |
111 |     async fn load_package(
    |              ^^^^^^^^^^^^

   Compiling parse_arg v0.1.4
error[E0599]: no method named `upsert` found for struct `Arc<warg_client::Client<FileSystemRegistryStorage, FileSystemContentStorage, FileSystemNamespaceMapStorage>>` in the current scope
   --> /home/tpmccallum/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-component-core-0.11.0/src/registry.rs:594:32
    |
594 |                 (index, client.upsert(upserts.iter()).await)
    |                                ^^^^^^ method not found in `Arc<Client<FileSystemRegistryStorage, FileSystemContentStorage, FileSystemNamespaceMapStorage>>`

Some errors have detailed explanations: E0061, E0308, E0599.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `cargo-component-core` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-component v0.11.0`, intermediate artifacts can be found at `/tmp/cargo-installvGJufU`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/README.md
+++ b/README.md
@@ -39,7 +39,7 @@ To install the `cargo component` subcommand from source, run the following
 command:
 
 ```
-cargo install cargo-component
+cargo install cargo-component --locked
 ```
 
 If you have the [cargo-binstall](https://github.com/cargo-bins/cargo-binstall)

